Highrachy/ballers-api
# #172304633 Refactor Register User API [User Activation - Part 1]

#### What does this PR do?
This PR creates a basic mailer template for ballers endpoints. It also creates a basic endpoint to view the templates internally (not meant for the public, hence the route details are not provided in the API doc).

#### Description of Task to be completed?
1. Create a basic HTML template for sending mails 
2. Create a basic text template for sending mails 
3. Create mailer service to generate the HTML and text template from provided options
4. Create routes a basic route`http://localhost:3000/mailer/templates` to view created templates.

#### How should this be manually tested?
1. Check out into this branch
2. Run `npm install` to update dependencies
3. Start the API server on your local machine `npm run start:dev`
4. Visit `http://localhost:3000/mailer/templates` to view the default template. If you are viewing this on POSTMAN, click on Preview to see the generated template
5. Change the type params to `text` (http://localhost:3000/mailer/templates?type=text).
6. The generated content is based on the value in `/mailers/index.js`. You can duplicate the DEFAULT object, give it a new name (e.g ACTIVATE_YOUR_ACCOUNT) and update the properties in the duplicated object. This can be viewed from the endpoint by using the name params. `http://localhost:3000/mailer/templates?name=ACTIVATE_YOUR_ACCOUNT`.

#### Screenshots
###### HTML Template
<img width="1137" alt="Screenshot 2020-04-16 at 1 06 54 PM" src="https://user-images.githubusercontent.com/26963369/79454605-6f4f6780-7fe3-11ea-98bf-b3fb90d0ba34.png">

###### Text Template
<img width="1133" alt="Screenshot 2020-04-16 at 1 07 36 PM" src="https://user-images.githubusercontent.com/26963369/79454641-7fffdd80-7fe3-11ea-91c9-b73b5fea0fb1.png">
